### ISSUE TITLE
Add support for SDPA to NLLB in Huggingface Transformers

### ISSUE THREAD

**Issue by ddaspit** [sillsdev/silnlp#478]
NLLB currently supports FlashAttention in HF Transformers. Unfortunately, FlashAttention results in degradation of quality, because it does not properly support [padding masks](https://github.com/Dao-AILab/flash-attention/issues/17). [SDPA](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) provides an alternative route for applying attention optimizations. Under the hood, it supports FlashAttention and Memory Efficient Attention. Memory Efficient Attention should support masking. [Here](https://github.com/huggingface/transformers/issues/28005) is the issue for adding SDPA support to models in Transformers. For a list of currently supported models, check out the Transformers [documentation](https://huggingface.co/docs/transformers/v4.44.0/en/perf_infer_gpu_one#pytorch-scaled-dot-product-attention). A good example to follow would be BART, which has a full encoder-decoder architecture. It might also be useful to check out this [PR](https://github.com/huggingface/transformers/pull/30375) that adds SDPA support to T5, another encoder-decoder LLM.

**Comment by isaac091** [comment]
PR submitted to transformers library last week, waiting on review.

**Comment by ddaspit** [comment]
Here is the PR: https://github.com/huggingface/transformers/pull/33309

**Comment by ddaspit** [comment]
That is awesome. Good job.
